spec/requests/api/v2/courses_spec.rb

```diff
@@ -0,0 +1,86 @@
+require "rails_helper"
+
+describe "Courses API v2", type: :request do
+  let(:user)         { create(:user) }
+  let(:organisation) { create(:organisation, users: [user]) }
+  let(:payload)      { { email: user.email } }
+  let(:token)        { build_jwt :apiv2, payload: payload }
+  let(:credentials) do
+    ActionController::HttpAuthentication::Token.encode_credentials(token)
+  end
+  let(:current_cycle) { find_or_create :recruitment_cycle }
+  let(:next_cycle)    { find_or_create :recruitment_cycle, :next }
+  let(:current_year)  { current_cycle.year.to_i }
+  let(:previous_year) { current_year - 1 }
+  let(:next_year)     { current_year + 1 }
+  let(:subjects) { [course_subject_mathematics] }
+
+  let(:providerA)       { create :provider, organisations: [organisation] }
+  let(:providerB)       { create :provider, organisations: [organisation] }
+  let(:accredited_body) { create :provider, :accredited_body }
+
+  let(:courseA) { create(:course, provider: providerA) }
+  let(:courseB) { create(:course, provider: providerB) }
+  let(:courseC) { create(:course) }
+  let(:courseD) { create(:course, provider: providerA, accrediting_provider: accredited_body) }
+
+  describe "GET index" do
+    subject { perform_request(request_path) }
+
+    def perform_request(path)
+      courseA
+      courseB
+      courseC
+      courseD
+      get path,
+          headers: { "HTTP_AUTHORIZATION" => credentials }
+      response
+    end
+
+    context "request without filters" do
+      let(:request_path) do
+        api_v2_recruitment_cycle_courses_path(current_cycle.year) +
+          "?include=subjects"
+      end
+
+      it { should have_http_status(:success) }
```

Comment:
personally i would merge this test into the test below. as these tests are expensive and there's no real gain to have it split out. it'll save the test setup and request